### ISSUE TITLE
how to Start Using Bash IDE

### ISSUE THREAD

**Issue by rodiaua** [mads-hartmann/bash-language-server#183]
Please explain how to start using Bash IDE in VSCode.
Please provide screenshots if it's possible.

**Comment by flandraco** [comment]
There are a few different possibilities: would you like to use Bash IDE on a Windows Platform, with WSL remote? without WSL remote?
Or just on a Linux Platform?

**Comment by flandraco** [comment]
Either way, it is important to understand that the Bash IDE 1.4.0 plugin is a CLIENT. In order to provide the full functionality, you need to install a language SERVER. And then you need to show how your CLIENT can find that SERVER. This will enable things such as outlining, code completion, reference search,...

If you want to actually debug your Bash Code, you need a different plugin (Bash Debug 0.3.7)

**Comment by rodiaua** [comment]
> There are a few different possibilities: would you like to use Bash IDE on a Windows Platform, with WSL remote? without WSL remote?
> Or just on a Linux Platform?

I am going to use bash ide on a windows. 
Appreciate your answers.

**Comment by rcjsuen** [comment]
@rodiaua Have you gone through the [`README.md`](https://github.com/mads-hartmann/bash-language-server/blob/e8451fd402e40e5955f87af359bd9f06a43153d5/README.md) file and decided which client you want to use? Or are you interested in using a text editor that is not listed there?

**Comment by flandraco** [comment]
The installation of Bash IDE should be straight forward.
The installation of the bash-language-server has a lot of methods, but only a few lead to results.
Background information:
The bash language server is installed with the 'npm' package manager. This package manager is available in the 'node.js' JavaScript runtime. I found the best way to get access to this, is through the full runtime:

1. Download 12.16.1 LTS from https://nodejs.org/en/.
2. Install the 'node.js' environment, 
2.1 Make sure that you have all the tools you install also accessible through your PATH:
![image](https://user-images.githubusercontent.com/10713636/75195033-0c8dce00-570e-11ea-9c47-f92efc162079.png)
2.2 I find the installation is best done by selecting the option to Automatically install the necessary tools (this includes MS Build Tools, Python, and Chocolatey):
![image](https://user-images.githubusercontent.com/10713636/75194810-abfe9100-570d-11ea-94d4-f32a0f7ae2ca.png)
3. When the installation is done (I seem to remember it took a while), there shouldn't be any errors; Proceed to install the server:
`npm i -g bash-language-server
`
Ii: install
-g: accessible to all users (I remember things go wrong if you don't do this)

4. Go to your Bash IDE settings, and provide the full path to the language server:
![image](https://user-images.githubusercontent.com/10713636/75195600-15cb6a80-570f-11ea-99df-4bb7b54163c0.png)

This is to the best of my recollection, please confirm if I missed anything?

**Comment by flandraco** [comment]
> @rodiaua Have you gone through the [`README.md`](https://github.com/mads-hartmann/bash-language-server/blob/e8451fd402e40e5955f87af359bd9f06a43153d5/README.md) file and decided which client you want to use? Or are you interested in using a text editor that is not listed there?

The initial question lists VS Code, so that's what I went ahead and explained. The actual installation of the bash-language-server should be common to all clients, I hope?

**Comment by rcjsuen** [comment]
> @rodiaua Have you gone through the README.md file and decided which client you want to use? Or are you interested in using a text editor that is not listed there?
> 
> The initial question lists VS Code, so that's what I went ahead and explained.

You are right, apologies for missing this.

> The actual installation of the bash-language-server should be common to all clients, I hope?

You are correct, this shouldn't change.

**Comment by skovhus** [comment]
> The initial question lists VS Code, so that's what I went ahead and explained. The actual installation of the bash-language-server should be common to all clients, I hope?

The language server is baked into the VS Code client/extension. So no additional installation steps are necessary.

**Comment by beltaurus** [comment]
> > The initial question lists VS Code, so that's what I went ahead and explained. The actual installation of the bash-language-server should be common to all clients, I hope?
> 
> The language server is baked into the VS Code client/extension. So no additional installation steps are necessary.

That is good to know. I just tested this by uninstalling bash-language-server from my Native Windows environment (and all of node.js while I was at it). The features are still working indeed! So the question is why the client/extension is still asking for a location of the binary (I am leaving it blank now):
![image](https://user-images.githubusercontent.com/54114043/75399066-2ebf5180-58b0-11ea-8084-5c04fbd82d1a.png)

I also uninstalled bash-language-server and node.js from my WSL environment. Same happy conclusion: the features are still there. But also same question, why still ask for the location:
![image](https://user-images.githubusercontent.com/54114043/75399423-0dab3080-58b1-11ea-85db-640a4fc182bf.png)
